Code/Tools/TestImpactFramework/Runtime/Common/Code/Include/Static/Dependency/TestImpactTestSelectorAndPrioritizer.h

```diff
@@ -351,16 +353,88 @@ namespace TestImpact
 
     template<typename ProductionTarget, typename TestTarget>
     AZStd::vector<const TestTarget*> TestSelectorAndPrioritizer<ProductionTarget, TestTarget>::PrioritizeSelectedTestTargets(
-        const SelectedTestTargetAndDependerMap& selectedTestTargetAndDependerMap,
-        [[maybe_unused]] Policy::TestPrioritization testSelectionStrategy)
+        const SelectedTestTargetAndDependerMap& selectedTestTargetAndDependerMap, const Policy::TestPrioritization testSelectionStrategy)
     {
         AZStd::vector<const TestTarget*> selectedTestTargets;
+        selectedTestTargets.reserve(selectedTestTargetAndDependerMap.size());
+
+        const auto fillSelectedTestTargets = [&selectedTestTargets](const auto& containerOfPairs)
+        {
+            AZStd::transform(
+                containerOfPairs.begin(),
+                containerOfPairs.end(),
+                AZStd::back_inserter(selectedTestTargets),
+                [](const auto& pair)
+                {
+                    return pair.first;
+                });
+        };
+
+        if (testSelectionStrategy == Policy::TestPrioritization::DependencyLocality)
+        {
+            const BuildTargetList<ProductionTarget, TestTarget>* buildTargetList = m_dynamicDependencyMap->GetBuildTargetList();
+            const BuildGraph<ProductionTarget, TestTarget>& buildGraph = buildTargetList->GetBuildGraph();
+
+            AZStd::vector<AZStd::pair<const TestTarget*, AZStd::optional<AZStd::size_t>>> testTargetDistancePairs;
+            testTargetDistancePairs.reserve(selectedTestTargetAndDependerMap.size());
+
+            // Loop through each test target in our map, walk the build dependency graph for that target and retrieve the vertices for the
+            // production targets in productionTargets
+            for (const auto [testTarget, productionTargets] : selectedTestTargetAndDependerMap)
+            {
+                AZStd::optional<AZStd::size_t> minimum;
+                buildGraph.WalkBuildDependencies(
+                    buildTargetList->GetBuildTargetOrThrow(testTarget->GetName()),
+                    [&](const BuildGraphVertex<ProductionTarget, TestTarget>& vertex, AZStd::size_t distance)
+                    {
+                        if (const auto productionTarget = vertex.m_buildTarget.GetProductionTarget();
+                            productionTarget && productionTargets.contains(productionTarget))
+                        {
+                            AZ_Printf("", "raaa");
```

Comment:
Just noticed this which I apparently didn't spot last time I reviewed this - is this meant to be here or is some left over debugging or something?

Reply:
Left over from debugging. Just removed as I saw it at the same time you did!

Reply:
👍